### PR TITLE
Ignore mir tests on big-endian

### PR DESCRIPTION
**Pull request by infinity0**
Other mir tests are already ignored on big-endian, e.g. const_allocation*.rs

### CODE REVIEW

File: src/test/mir-opt/const_prop/tuple_literal_propagation.main.ConstProp.diff [hunk 1]
```diff
@@ -2,40 +2,40 @@
...
++         _3 = const (1_u32, 2_u32);       // scope 1 at $DIR/tuple_literal_propagation.rs:6:13: 6:14
 +                                          // ty::Const
 +                                          // + ty: (u32, u32)
 +                                          // + val: Value(ByRef { alloc: Allocation { bytes: [1, 0, 0, 0, 2, 0, 0, 0], relocations: Relocations(SortedMap { data: [] }), init_mask: InitMask { blocks: [255], len: Size { raw: 8 } }, size: Size { raw: 8 }, align: Align { pow2: 2 }, mutability: Not, extra: () }, offset: Size { raw: 0 } })
 +                                          // mir::Constant
-+                                          // + span: $DIR/tuple_literal_propagation.rs:5:13: 5:14
++                                          // + span: $DIR/tuple_literal_propagation.rs:6:13: 6:14
 +                                          // + literal: Const { ty: (u32, u32), val: Value(ByRef { alloc: Allocation { bytes: [1, 0, 0, 0, 2, 0, 0, 0], relocations: Relocations(SortedMap { data: [] }), init_mask: InitMask { blocks: [255], len: Size { raw: 8 } }, size: Size { raw: 8 }, align: Align { pow2: 2 }, mutability: Not, extra: () }, offset: Size { raw: 0 } }) }
```
Comment (Mark-Simulacrum) [inline src/test/mir-opt/const_prop/tuple_literal_propagation.main.ConstProp.diff:26]: This seems to be a "significant" diff -- i.e., not just line numbers changing. Maybe a bug in how github is showing it or a rebase error?

**Comment by Mark-Simulacrum** [comment]
I assume these are failing on big endian platforms?

I think the const allocation ones are ignored because the output contains printed allocations (e.g., https://github.com/rust-lang/rust/blob/738d4a7a368b4da8bfa31915f06bb1ddce9e9980/src/test/mir-opt/const_allocation2.main.ConstProp.after.32bit.mir#L31-L33) which are not normalized to a particular endian-ness currently (probably no need to do so either).

But the tests here, aside from the one comment I left, don't seem to contain such allocations. I am a bit worried that we could accidentally mask some real difference in const prop or whatever on big endian systems. If we expect to just ignore all of the MIR tests then maybe doing that in compiletest makes more sense.

cc @oli-obk @ecstatic-morse (not sure who all to cc on const prop/MIR, would be good to get a team)

**Comment by ecstatic-morse** [comment]
cc @rust-lang/wg-mir-opt

File: src/test/mir-opt/const_prop/tuple_literal_propagation.main.ConstProp.diff (hunk 1 above)
Comment (ecstatic-morse) [reply src/test/mir-opt/const_prop/tuple_literal_propagation.main.ConstProp.diff:26]: This is just line numbers changing AFAICT. Diffs of diffs are hard to parse for me also.

**Comment by ecstatic-morse** [comment]
@infinity0 It's helpful to reviewers to give some motivation for a PR. I assume you're running on a big-endian system and that these tests are failing due to blessed MIR like:

```
+         _1 = const (4_u32, false);       // scope 0 at $DIR/return_place.rs:6:5: 6:10
+                                          // ty::Const
+                                          // + ty: (u32, bool)
+                                          // + val: Value(ByRef { alloc: Allocation { bytes: [4, 0, 0, 0, 0, 0, 0, 0], relocations: Relocations(SortedMap { data: [] }), init_mask: InitMask { blocks: [31], len: Size { raw: 8 } }, size: Size { raw: 8 }, align: Align { pow2: 2 }, mutability: Not, extra: () }, offset: Size { raw: 0 } })
+                                          // mir::Constant
+                                          // + span: $DIR/return_place.rs:6:5: 6:10
+                                          // + literal: Const { ty: (u32, bool), val: Value(ByRef { alloc: Allocation { bytes: [4, 0, 0, 0, 0, 0, 0, 0], relocations: Relocations(SortedMap { data: [] }), init_mask: InitMask { blocks: [31], len: Size { raw: 8 } }, size: Size { raw: 8 }, align: Align { pow2: 2 }, mutability: Not, extra: () }, offset: Size { raw: 0 } }) }
```

Normally I would request that we normalize MIR dumps across different systems, but this is a) kind of hard and b) not the precedent (as you note in the OP). I defer to Oli on this one.

r? @oli-obk

File: src/test/mir-opt/const_prop/tuple_literal_propagation.main.ConstProp.diff (hunk 1 above)
Comment (oli-obk) [reply src/test/mir-opt/const_prop/tuple_literal_propagation.main.ConstProp.diff:26]: Things like these can be avoided by adding new comment lines in existing empty lines or just adding them at the end of the file. We have no policy on this though

File: src/test/mir-opt/const_prop/tuple_literal_propagation.rs [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// ignore-endian-big
```
Comment (oli-obk) [inline src/test/mir-opt/const_prop/tuple_literal_propagation.rs:1]: Instead of ignoring this test we should add tuples and arrays to the "don't verbosely print the constant" list if all tuple fields or the array element type are on the "don't verbosely print constants of this type" list.

File: src/test/mir-opt/const_prop/mutable_variable_no_prop.rs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 // compile-flags: -O
-
+// ignore-endian-big
```
Comment (oli-obk) [inline src/test/mir-opt/const_prop/mutable_variable_no_prop.rs:2]: This one is odd, so I think it's ok to ignore here

**Comment by oli-obk** [comment]
So imo only `src/test/mir-opt/const_prop/mutable_variable_no_prop.rs` should be ignored, everything else can be fixed

File: src/test/mir-opt/const_prop/tuple_literal_propagation.main.ConstProp.diff (hunk 1 above)
Comment (infinity0) [reply src/test/mir-opt/const_prop/tuple_literal_propagation.main.ConstProp.diff:26]: Yes this is just line numbers changing - the existing file didn't have an empty line to fit a comment into, and I thought putting it at the end of the file would be inconsistent with everything else.

**Comment by infinity0** [comment]
> we should add tuples and arrays to the "don't verbosely print the constant" list if all tuple fields or the array element type are on the "don't verbosely print constants of this type" list.

Shall I file an issue for this then? I'm not sufficiently involved with MIR to make this change myself. In the meantime do you want to keep the tests failing?

**Comment by oli-obk** [comment]
Oh, this is not related to MIR, just pretty printing. All you have to do is to pull out https://github.com/rust-lang/rust/blob/5ded394553296d56bb66e925d7001ab3271979ce/compiler/rustc_mir/src/util/pretty.rs#L432-L436 into a separate function that returns `true` if the verbose printing should be used and then make it recurse on `ty::Tuple` and `ty::Array` elements

**Comment by crlf0710** [comment]
@infinity0 Ping from triage! Any updates on this?

**Comment by JohnCSimon** [comment]
@infinity0 Pinging again from triage! Any updates on this?

**Comment by infinity0** [comment]
Sorry, I don't have the time to do the alternative. But the underlying problem exists and should be fixed.

**Comment by oli-obk** [comment]
I'll open an issue and see if someone picks it up. Thanks for bringing up the issue!
